src/lib/y2network/interface_config_builder.rb

```diff
@@ -262,27 +262,31 @@ def driver=(value)
     end
 
     # gets aliases for interface
-    # @return [Array<Hash>] hash values are `:label` for alias label,
-    #   `:ip` for ip address, `:mask` for netmask and `:prefixlen` for prefix.
-    #   Only one of `:mask` and `:prefixlen` is set.
+    # @return [Array<Hash>] array of the connection additional IP address in
+    #   hash format see #alias_for for the hash values
     def aliases
       return @aliases if @aliases
 
-      aliases = @connection_config.ip_aliases.map do |data|
-        {
-          label:     data.label.to_s,
-          ip:        data.address.address.to_s,
-          prefixlen: data.address.prefix.to_s,
-          id:        data.id.to_s
-          # NOTE: new API does not have netmask at all, we need to adapt UI to clearly mention
-          #       only prefix
-        }
-      end
-      @aliases = aliases
+      @aliases = @connection_config.ip_aliases.map { |d| alias_for(d) }
+    end
+
+    # Convenience method to obtain a new hash from the IP additional address
+    # data
+    #
+    # @param data [IPConfig, nil] Additional IP address configuration
+    # @return [Hash<String>] hash values are `:label` for alias label,
+    #   `:ip_address` for ip address, `:mask` for netmask and `:subnet_prefix` for prefix.
+    def alias_for(data)
+      {
+        label:         data&.label.to_s,
+        ip_address:    data&.address&.address.to_s,
+        subnet_prefix: data&.address&.prefix.to_s,
```

Comment:
Probably better to do as we are expecting an "/" when converting back.

```suggestion
        subnet_prefix: data ? "/#{data.prefix.to_s}" : ""
```